AsyncStartTest#shouldWaitForAction failed on jenkins once

Saw this once on jenkins

https://hq2.camunda.com/jenkins/zeebe/job/zeebe/job/975-customevent-test/2/console

```java
18:39:19 [ERROR] Tests run: 15, Failures: 1, Errors: 0, Skipped: 3, Time elapsed: 0.02 s <<< FAILURE! - in io.zeebe.servicecontainer.impl.AsyncStartTest
18:39:19 [ERROR] shouldWaitForAction(io.zeebe.servicecontainer.impl.AsyncStartTest)  Time elapsed: 0.006 s  <<< FAILURE!
18:39:19 java.lang.AssertionError: 
18:39:19 
18:39:19 Expecting
18:39:19   <CompletableActorFuture{value= java.lang.Object@6eb2384f}>
18:39:19 not to be done
18:39:19 	at io.zeebe.servicecontainer.impl.ActorFutureAssertions.assertNotCompleted(ActorFutureAssertions.java:28)
18:39:19 	at io.zeebe.servicecontainer.impl.AsyncStartTest.shouldWaitForAction(AsyncStartTest.java:148)
18:39:19 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
18:39:19 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
18:39:19 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
18:39:19 	at java.lang.reflect.Method.invoke(Method.java:498)
18:39:19 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
18:39:19 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
18:39:19 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
18:39:19 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
18:39:19 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
18:39:19 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
18:39:19 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
18:39:19 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
18:39:19 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
18:39:19 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
18:39:19 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
18:39:19 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
18:39:19 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
18:39:19 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
18:39:19 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
18:39:19 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
18:39:19 	at org.junit.runners.Suite.runChild(Suite.java:128)
18:39:19 	at org.junit.runners.Suite.runChild(Suite.java:27)
18:39:19 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
18:39:19 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
18:39:19 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
18:39:19 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
18:39:19 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
18:39:19 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
18:39:19 	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
18:39:19 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
18:39:19 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
18:39:19 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
18:39:19 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
18:39:19 	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:157)
18:39:19 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
18:39:19 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
18:39:19 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
18:39:19 
```

Can't reproduce it locally, run it in a loop for one hour or more

--- a/service-container/src/test/java/io/zeebe/servicecontainer/impl/AsyncStartTest.java
+++ b/service-container/src/test/java/io/zeebe/servicecontainer/impl/AsyncStartTest.java
@@ -15,15 +15,25 @@
  */
 package io.zeebe.servicecontainer.impl;
 
-import static io.zeebe.servicecontainer.impl.ActorFutureAssertions.*;
+import static io.zeebe.servicecontainer.impl.ActorFutureAssertions.assertCompleted;
+import static io.zeebe.servicecontainer.impl.ActorFutureAssertions.assertFailed;
+import static io.zeebe.servicecontainer.impl.ActorFutureAssertions.assertNotCompleted;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.Assert.assertEquals;
-import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
 
-import io.zeebe.servicecontainer.*;
+import io.zeebe.servicecontainer.Service;
+import io.zeebe.servicecontainer.ServiceContainer;
+import io.zeebe.servicecontainer.ServiceName;
+import io.zeebe.servicecontainer.ServiceStartContext;
+import io.zeebe.servicecontainer.ServiceStopContext;
 import io.zeebe.util.sched.future.ActorFuture;
 import io.zeebe.util.sched.future.CompletableActorFuture;
 import io.zeebe.util.sched.testing.ControlledActorSchedulerRule;
+import java.util.concurrent.BrokenBarrierException;
+import java.util.concurrent.CyclicBarrier;
 import org.junit.Before;
 import org.junit.Ignore;
 import org.junit.Rule;
@@ -137,8 +147,7 @@ public class AsyncStartTest {
   public void shouldWaitForAction() {
     // when
     final AsyncStartService service = new AsyncStartService();
-    final Runnable mockAction = mock(Runnable.class);
-    service.action = mockAction;
+    service.action = new BlockingAction();
 
     final ActorFuture<Object> startFuture =
         serviceContainer.createService(service1Name, service).install();
@@ -149,19 +158,19 @@ public class AsyncStartTest {
   }
 
   @Test
-  public void shouldContinueOnAction() {
+  public void shouldContinueOnAction() throws BrokenBarrierException, InterruptedException {
     // given
     final AsyncStartService service = new AsyncStartService();
-    final Runnable mockAction = mock(Runnable.class);
-    service.action = mockAction;
+    final BlockingAction action = new BlockingAction();
+    service.action = action;
 
     final ActorFuture<Object> startFuture =
         serviceContainer.createService(service1Name, service).install();
     actorSchedulerRule.workUntilDone();
-    actorSchedulerRule.awaitBlockingTasksCompleted(1);
 
     // when
-    verify(mockAction).run();
+    action.signal();
+    actorSchedulerRule.awaitBlockingTasksCompleted(1);
     actorSchedulerRule.workUntilDone();
 
     // then
@@ -379,6 +388,25 @@ public class AsyncStartTest {
     @Override
     public Object get() {
       return value;
+    }
+  }
+
+  /** Runnable which blocks until signaled */
+  static class BlockingAction implements Runnable {
+
+    CyclicBarrier barrier = new CyclicBarrier(2);
+
+    @Override
+    public void run() {
+      try {
+        barrier.await();
+      } catch (InterruptedException | BrokenBarrierException e) {
+        e.printStackTrace();
+      }
+    }
+
+    public void signal() throws BrokenBarrierException, InterruptedException {
+      barrier.await();
     }
   }
 }

